### PR TITLE
Support for jaeger remote sampler

### LINKED ISSUE #36335: quarkus.otel.traces.sampler.arg doesn't support String value (e.g. cannot configure "jaeger_remote" sampler)
### Describe the bug

The [Quarkus OpenTelemetry Configuration Reference](https://quarkus.io/guides/opentelemetry#configuration-reference) mentions that the `quarkus.*` properties match the ones from the [OpenTelemetry SDK Autoconfigure](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure/README.md).

In the AutoConfigure SDK [sampler](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure/README.md#sampler) configuration section, the `otel.traces.sampler.arg` property isn't explicitly restricted to having a number/double value.
In fact, in the OpenTelemetry Specification 1.25.0, section [General SDK Configuration](https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#general-sdk-configuration), the `OTEL_TRACES_SAMPLER_ARG` env var is described as possibly containing a comma separated list in case of the `jaeger_remote` and `parentbased_jaeger_remote` samplers.

But the [quarkus.otel.traces.sampler.arg](https://quarkus.io/guides/opentelemetry#quarkus-opentelemetry_quarkus.otel.traces.sampler.arg) configuration property explicitly has a `double` type.

That means that a configuration of 
```
quarkus.otel.traces.sampler: "jaeger_remote"
quarkus.otel.traces.sampler.arg: "endpoint=http://hono-jaeger:14250"
```
isn't supported and leads to this exception on startup:
```
ERROR: Failed to start application (with profile [prod])
java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
	at io.quarkus.runtime.Application.start(Application.java:101)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:111)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:61)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:32)
Caused by: io.smallrye.config.ConfigValidationException: Configuration validation failed:
	java.lang.IllegalArgumentException: SRCFG00039: The config property quarkus.otel.traces.sampler.arg with the config value "endpoint=http://hono-jaeger:14250" threw an Exception whilst being converted SRCFG00031: Expected a double value, got "endpoint=http://hono-jaeger:14250"
	at io.smallrye.config.ConfigMappingProvider.mapConfigurationInternal(ConfigMappingProvider.java:1003)
	at io.smallrye.config.ConfigMappingProvider.lambda$mapConfiguration$3(ConfigMappingProvider.java:959)
	at io.smallrye.config.SecretKeys.doUnlocked(SecretKeys.java:28)
	at io.smallrye.config.ConfigMappingProvider.mapConfiguration(ConfigMappingProvider.java:959)
	at io.smallrye.config.ConfigMappings.mapConfiguration(ConfigMappings.java:91)
	at io.smallrye.config.SmallRyeConfigBuilder.build(SmallRyeConfigBuilder.java:630)
	at io.quarkus.runtime.generated.Config.readConfig(Unknown Source)
	at io.quarkus.deployment.steps.RuntimeConfigSetup.deploy(Unknown Source)
	... 13 more
```

### Expected behavior

`quarkus.otel.traces.sampler.arg` not being restricted to a `double` type.

### Actual behavior

Setting `quarkus.otel.traces.sampler.arg` to a String value leads to an exception on startup.

### How to Reproduce?

Use a configuration like
```
quarkus.otel.traces.sampler: "jaeger_remote"
quarkus.otel.traces.sampler.arg: "endpoint=http://hono-jaeger:14250"
```

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

3.2.6

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by hiteshkhatri97**
Fixes #36335

### CODE REVIEW

**Comment by hiteshkhatri97** [comment]
This is part of the ongoing hacktoberfest. Please add the label "hacktoberfest-accepted" before merging

**Comment by hiteshkhatri97** [comment]
@brunobat @radcortez please review

**Review by brunobat — CHANGES_REQUESTED** [review]
@hiteshkhatri97 The `opentelemetry-sdk-extension-jaeger-remote-sampler` must not be part of the main extension. 
It's fine to change the config values to string but you need to handle the consequences o the current default sampler.
On the OpenTelemetry Integration test, you can include `opentelemetry-sdk-extension-jaeger-remote-sampler`  and create a test in there to see if it works.

**Comment by hiteshkhatri97** [comment]
@brunobat are you saying `opentelemetry-sdk-extension-jaeger-remote-sampler` is not required in this case or we shouldn't add support for this? Sorry, I'm new and just looking for some clarification.

**Comment by hiteshkhatri97** [comment]
If I understand correctly, if someone wants to configure the jaeger_remote sampler, they must include its dependency in their project.

**Comment by hiteshkhatri97** [comment]
Also, just wondering if integration tests are required then they should be part of this PR or separate PR?

**Comment by brunobat** [comment]
Tests must be part of the PR.
The main point of the PR, in my opinion, is to allow the `arg` to hold a String value and deal with potential side effects on the current code.
After this change, other samplers can take benefit of the String 'arg'. To prove this is the case, you can create an integration test that will use `opentelemetry-sdk-extension-jaeger-remote-sampler`.

**Comment by hiteshkhatri97** [comment]
@brunobat This is ready for early review.

File: integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentracing/OTelJaegerRemoteTest.java
Comment (brunobat) [inline integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentracing/OTelJaegerRemoteTest.java:38]: please build the project for the automatic formater to kick in... Static attributes must be declared before injections.

File: integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentracing/OTelJaegerRemoteTest.java
Comment (brunobat) [inline integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentracing/OTelJaegerRemoteTest.java:84]: Please take a look on how to create manual spans, here: https://opentelemetry.io/docs/instrumentation/java/manual/#create-spans
try/finally block is needed with an OTel Scope.
You don't need the sleep.

**Review by brunobat — CHANGES_REQUESTED** [review]

File: integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentracing/OTelJaegerRemoteTest.java
Comment (hiteshkhatri97) [reply integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentracing/OTelJaegerRemoteTest.java:38]: @brunobat I built the project using the command mvn process-sources as seen in the doc for formatting but I didn't get any errors. Can you please share the mvn command or ref in the doc for formatting?

Comment (brunobat) [reply integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentracing/OTelJaegerRemoteTest.java:38]: The `mvn clean install` will do it automatically. The recommendation I did still applies: static before inject

File: integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentracing/OTelJaegerRemoteTest.java
Comment (hiteshkhatri97) [reply integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentracing/OTelJaegerRemoteTest.java:84]: Thanks for the review @brunobat. I'm working on the other things, meanwhile, I'm kind of stuck on one issue. Basically, I'm trying to test Jaeger autoconfiguration and it needs a docker container to set the exporter endpoint correctly. So I'm creating a test container in the tests but since the configurations are set before the initialization of the container in the test, it is not able to set the exporter properly, resulting in flaky tests. Can you guide what could be done here?

File: integration-tests/opentelemetry-jaeger-remote/pom.xml
Comment (hiteshkhatri97) [reply integration-tests/opentelemetry-jaeger-remote/pom.xml]: Done

File: integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentracing/OTelJaegerRemoteTest.java
Comment (hiteshkhatri97) [reply integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentracing/OTelJaegerRemoteTest.java]: Done

File: integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentracing/OTelJaegerRemoteTest.java
Comment (hiteshkhatri97) [reply integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentracing/OTelJaegerRemoteTest.java]: Have done changes after referring this.

**Comment by brunobat** [comment]
[hiteshkhatri97](https://github.com/hiteshkhatri97) will checkout the project and see what happens with the test.

**Comment by hiteshkhatri97** [comment]
Thanks @brunobat. Now the tests look fine to me. I had kept `initialSamplingRate` very low. I have changed it now.

**Comment by brunobat** [comment]
Excelent, will approve the CI run and see how it goes. BTW, we require all commits to be squashed into a single commit per PR.

**Comment by hiteshkhatri97** [comment]
Oh, I thought commits could be squashed while merging the PR. Nvm I will do it

**Comment by hiteshkhatri97** [comment]
@brunobat commits are squashed. Let me know what you think

**Comment by hiteshkhatri97** [comment]
@brunobat can we possibly try to close this today if everything looks good?

File: integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentracing/OTelJaegerRemoteTest.java
Comment (brunobat) [inline integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentracing/OTelJaegerRemoteTest.java:1]: Sorry but this is opentelemetry, not opentracing.

File: integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentracing/OTelJaegerRemoteTest.java
Comment (brunobat) [inline integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentracing/OTelJaegerRemoteTest.java:45]: This static should be above inject

File: integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentracing/OTelJaegerRemoteTest.java
Comment (brunobat) [inline integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentracing/OTelJaegerRemoteTest.java:105]: Couldn't you just use restasured?
There's many examples on other OTel tests that can make this test more legible and maintainable... No need to be parsing json.

File: integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentracing/OTelJaegerRemoteTest.java
Comment (brunobat) [inline integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentracing/OTelJaegerRemoteTest.java:85]: This test will be flaky. You need to fire a few traces and count the ones arriving must be equal or above the expected sample. 
You have a sample rate of 0,9, meaning, this test, like it is now will fail  1 each 10 runs

**Review by brunobat — CHANGES_REQUESTED** [review]
Sorry @hiteshkhatri97, took a deep look at the test and I think it needs to be improved.

File: integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentracing/OTelJaegerRemoteTest.java
Comment (hiteshkhatri97) [reply integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentracing/OTelJaegerRemoteTest.java:1]: Will change this

File: integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentracing/OTelJaegerRemoteTest.java
Comment (hiteshkhatri97) [reply integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentracing/OTelJaegerRemoteTest.java:85]: On the sampling rate, should I keep it to 1.0?

Comment (brunobat) [reply integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentracing/OTelJaegerRemoteTest.java:85]: You can but you should find a way to check that this sampler is actually being used.

Comment (hiteshkhatri97) [reply integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentracing/OTelJaegerRemoteTest.java:85]: One more thing that I didn't understand from your comment is if I fire a few traces then how will I get arriving above the expected sample? Wouldn't it be equal or less?

Comment (brunobat) [reply integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentracing/OTelJaegerRemoteTest.java:85]: I was thinking that if you fire 10 spans, with 0.9 sampling, only 9 should arrive... But this is not deterministic and therefore, also flaky...
I'd say, fire 1 span and make the sample arg be 1.0. Also find a way to know which sampler is being used... Most likely a different test, on the deployment tests, will be easier.

Comment (hiteshkhatri97) [reply integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentracing/OTelJaegerRemoteTest.java:85]: I got it now. Will try to incorporate changes in these tests only but if doesn't work then will create a separate one.

Comment (hiteshkhatri97) [reply integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentracing/OTelJaegerRemoteTest.java]: Have updated the tests

File: integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentelemetry/jaeger/OTelJaegerRemoteTest.java
Comment (brunobat) [inline integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentelemetry/jaeger/OTelJaegerRemoteTest.java:120]: we cannot have this assert or all releases will fail.

File: integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentelemetry/jaeger/OTelJaegerRemoteTest.java
Comment (brunobat) [inline integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentelemetry/jaeger/OTelJaegerRemoteTest.java:117]: Sorry @hiteshkhatri97 but you should remove this method. None of these assertions is really relevant to understand if the jaeger sampler is actually being used. 
Let's assume it is for now and move on.

**Review by brunobat — CHANGES_REQUESTED** [review]
Getting closer.

File: integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentelemetry/jaeger/OTelJaegerRemoteTest.java
Comment (hiteshkhatri97) [reply integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentelemetry/jaeger/OTelJaegerRemoteTest.java:120]: I saw it in some test but nvm I'm removing this.

File: integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentelemetry/jaeger/OTelJaegerRemoteTest.java
Comment (hiteshkhatri97) [reply integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentelemetry/jaeger/OTelJaegerRemoteTest.java]: updated

**Comment by hiteshkhatri97** [comment]
@brunobat Sorry I'm bugging you again but if you can take a look. Thanks

File: extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/config/runtime/TracesRuntimeConfig.java
Comment (brunobat) [inline extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/config/runtime/TracesRuntimeConfig.java:46]: We need to also clarify this documentation or people will mess up the config because we are no longer enforcing a double upfront.
It should read:
````
Sampler argument. Depends on the `quarkus.otel.traces.sampler` property.
Fallbacks to the legacy property <code>quarkus.opentelemetry.tracer.sampler.ratio</code>.
<p>
When setting the stock sampler to `traceidratio` or `parentbased_traceidratio` you need to set a `double` compatible value like `0.01d` or `0.5d`. It is kept as a `String` to allow the flexible customisation of alternative samplers.
<p>
Defaults to `1.0d`.
````

**Review by brunobat — CHANGES_REQUESTED** [review]
The test looks good.

File: extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/config/runtime/TracesRuntimeConfig.java
Comment (hiteshkhatri97) [reply extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/config/runtime/TracesRuntimeConfig.java:48]: Done

**Review by brunobat — APPROVED** [review]
Thanks @hiteshkhatri97 for your effort!

**Comment by hiteshkhatri97** [comment]
Thanks a lot, @brunobat for your review, help, and support. Will merge it once CI run completes if I have access to merge

**Comment by hiteshkhatri97** [comment]
@brunobat can you please merge it? Looks like I don't have access

**Comment by brunobat** [comment]
Usually that's up to @gsmet or @geoand

File: integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentelemetry/jaeger/OTelJaegerRemoteIT.java
Comment (geoand) [inline integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentelemetry/jaeger/OTelJaegerRemoteIT.java:5]: I assume this is meant to test things in native mode?

If so, then we'll need to configure the proper native JSON config file in the .github directory

**Review by geoand — COMMENTED** [review]
I added a comment

File: integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentelemetry/jaeger/OTelJaegerRemoteIT.java
Comment (brunobat) [reply integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentelemetry/jaeger/OTelJaegerRemoteIT.java:5]: This is quite specific. Probably no need. There are other IT related projects for OTel that are not running on Native.

Comment (geoand) [reply integration-tests/opentelemetry-jaeger-remote/src/test/java/io/quarkus/it/opentelemetry/jaeger/OTelJaegerRemoteIT.java:5]: 👌

**Comment by hiteshkhatri97** [comment]
@geoand/@gsmet Can you please merge it? Let me know if there are any concerns.

**Comment by hiteshkhatri97** [comment]
@geoand / @gsmet Can you please merge it? Let me know if there are any concerns.

**Comment by geoand** [comment]
@brunobat will take care of that

**Pull request merged**
